src/client/CodeEditor/getOrCreateEditor.tsx

```diff
@@ -454,6 +458,67 @@ export const getOrCreateEditor = async ({
     extensions.push(copilot({ aiCopilotEndpoint }));
   }
 
+  // Widget appears after instances of "todo" in code editor, allowing AI to implement todo tasks when clicked
+  // TODO make widget only appear in comments(?)
+  class ToDoWidget extends WidgetType {
+    constructor() { super() }
+
+    eq(other: ToDoWidget) { return false }
+
+    toDOM() {
+      const wrap = document.createElement("i");
+      wrap.style.display = "inline-flex";  // prevent block expansion
+      wrap.style.alignItems = "center";
+      wrap.style.justifyContent = "center";
+      wrap.className = "icon-button icon-button-dark";
+      const reactContainer = document.createElement("div");
+      wrap.appendChild(reactContainer);
+      const root = createRoot(reactContainer);
+      root.render(<SparklesSVG w={14} h={14}/>);
+      return wrap
+    }
+
+    ignoreEvent() { return false }
+  }
+
+  function toDoWidgets(editor: EditorView) {
+    const findToDo = new RegExpCursor(editor.state.doc, 'todo', { ignoreCase: true });
+    let widgets = []
+      while (!findToDo.next().done) {
+        const match = findToDo.value;
+        const deco = Decoration.widget({
+          widget: new ToDoWidget(),
+          side: 1
+        }).range(match.to);
+        widgets.push(deco);
+      }
+    return Decoration.set(widgets)
+  }
+
+  const checkboxPlugin = ViewPlugin.fromClass(class {
```

Comment:
leftover name from reference example

```suggestion
  const todoPlugin = ViewPlugin.fromClass(class {
```